tests/encfs/encfs.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+from contextlib import contextmanager
+import subprocess
+import tempfile
+
+
+class CryptSetupFileSystem:
+    DEVICE_NAME = "cryptdevice1"
+    DEVICE_NAME_PATH = f"/dev/mapper/{DEVICE_NAME}"
+
+    def _run_command(self, *args):
+        subprocess.check_call(
+            " ".join(
+                [
+                    "sudo cryptsetup --debug -v",
+                    *args,
+                ]
+            ),
+            shell=True,
+        )
+
+    def __init__(self, key_path, image_path):
+        self.key_path = key_path
+        self.image_path = image_path
+        with open(image_path, "wb") as f:
+            f.seek(64 * 1024 * 1024 - 1)
+            f.write(b"\0")
+
+    def __enter__(self):
+        # Format
+        self._run_command(
+            "luksFormat",
+            "--type luks2",
+            self.image_path,
+            "--key-file",
+            f'"{self.key_path}"',
+            "--batch-mode",
+            "--sector-size 4096",
+            "--cipher aes-xts-plain64",
+            "--pbkdf pbkdf2",
+            "--pbkdf-force-iterations 1000",
+        )
+
+        # Open
+        self._run_command(
+            "luksOpen",
+            self.image_path,
+            self.DEVICE_NAME,
+            "--key-file",
+            self.key_path,
+            # Don't use a journal to increase performance
+            "--integrity-no-journal",
+            "--persistent",
+        )
+
+        # Mount
+        subprocess.check_call(f"sudo mkfs.ext4 {self.DEVICE_NAME_PATH}", shell=True)
+        self._dir = tempfile.TemporaryDirectory()
+        subprocess.check_call(
+            f"sudo mount -t ext4 {self.DEVICE_NAME_PATH} {self._dir.name} -o loop",
+            shell=True,
+        )
+        return self._dir.name
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        subprocess.check_call(f"sudo umount {self._dir.name}", shell=True)
+        self._run_command("luksClose", self.DEVICE_NAME)
+        self._dir.cleanup()
```

Comment:
nit: maybe try-finally here to ensure `cleanup()` is always run

Reply:
I've done some refactoring to make cleanup more robust here bbd4a68db95b1ccc05a7e13191eb93812348fbb9